src/server/pfs/server/driver_v2.go

```diff
@@ -144,53 +144,60 @@ func (d *driverV2) getSubFileSet() int64 {
 	return n
 }
 
-// TODO Need commit validation and handling of branch names.
-func (d *driverV2) withUnorderedWriter(ctx context.Context, repo, commit string, cb func(*fileset.UnorderedWriter) error) (retErr error) {
+// withCommitWriter calls cb with an unordered writer. All data written to cb is added to the commit, or an error is returned.
+func (d *driverV2) withCommitWriter(ctx context.Context, repo, commit string, cb func(*fileset.UnorderedWriter) error) (retErr error) {
 	n := d.getSubFileSet()
 	subFileSetStr := fileset.SubFileSetStr(n)
 	subFileSetPath := path.Join(repo, commit, subFileSetStr)
-	fs, err := d.storage.New(ctx, path.Join(tmpPrefix, subFileSetPath), subFileSetStr)
+	id, pathsWritten, err := d.withTmpUnorderedWriter(ctx, false, cb)
 	if err != nil {
 		return err
 	}
+	// collect paths for renewal
+	tmpPath := path.Join(tmpRepo, id)
+	return d.storage.WithRenewal(ctx, defaultTTL, pathsWritten, func(ctx context.Context) error {
```

Comment:
I don't think this is going to accomplish what we want. A long upload could still result with temporary subfilesets getting deleted since we wouldn't renew until the upload is finished. I was thinking about this a bit, and I think a model we could use that would work across the board would be to have a `Renewer` abstraction that could be passed around and you would be able to make calls on it to kick off renewals for new paths. Then you could have a `Close` function on it that would cancel all of the renewals that have been setup. This means you could setup the `Renewer` in the outer scope and defer the close to be sure that all the renewal goroutines are cleaned up (it might even make sense to just have a `WithRenewer` function). So, how this would work in the unordered writer case is that a `Renewer` would be setup in the outer scope (`withCommitWriter` in this case) and the unordered writer would take the `Renewer` as a parameter and add new paths to renew as it serializes subfilesets. Then when the outer scope is finished, all the renewals that were setup are canceled. For compaction the parent node would setup a `Renewer` for the child nodes and pass it in. Then when the children finish and the parent compacts them, the `Renewer` would be closed which would cancel all of the renewals.